lstchain/reco/r0_to_dl1.py

```diff
@@ -510,8 +520,14 @@ def r0_to_dl1(
                                            calibration_index,
                                            event.mon.tel[tel_id],
                                            new_ped=new_ped_event, new_ff=new_ff_event)
-
+                    
+                    # write the calibrated R1 waveform without gain selection
+                    source.r0_r1_calibrator.select_gain = False
```

Comment:
I really don't like that you modify the r0_r1_calibrator of the source here. Also, isn't it too late for that here? The source already applied the calibration when it reaches lstchain.

Reply:
Max,  it works because we have the "calibrate_flatfields_and_pedestals": false, in the config file (by the way, this code works well, I could "Cat-B" calibrate the GRB data with it ;-) )
I agree that it is not nice to change the source here, but the source is in any case treating in a special way these events, which one day will be in a dedicated stream hopefully.

Reply:
@maxnoe do you prefer I make a copy of the  source.r0_r1_calibrator  in order to touch an independent object?

Reply:
With the fixed code, this line should not be needed, right?

Reply:
the fix was to avoid the re-calibration and just to perform the gain selection

Reply:
> But calibration events are not gain-selected, are they?
After they are used for creating the calibration coefficients, they are gain selected

Reply:
> Or should we add a new flag to the source to calibrate but not gain select the calibration events?

a new flag would be cleaner but we in any case the present r0_to_dl1 is something we want to improve as soon as possible, I let you decide if you want add this flag and change the source version now

Reply:
I would say that if @maxnoe has time to implement this right away, we could make a minor release of `ctapipe_io_lst`, otherwise @FrancaCassol could open an issue to remember to modify this in the future

Reply:
Hi @rlopezcoto, I can do it but I would like to recall that  in a (hopefully) short future the cat-B calibration will be no more performed in the r0_to_dl1 script and also the interleaved images will be no more written in the dl1 file but will follow a parallel path treated by the calibration pipeline. If @maxnoe has not time to implement it just now I would go ahead with this PR as it is

Reply:
sure, this was my suggestion, if it cannot be implemented right now, let's open an issue to remember this and move on with the PR

Reply:
Opened here:
https://github.com/cta-observatory/ctapipe_io_lst/issues/186

thx!